superset-frontend/src/dashboard/util/isFilterChart.js

```diff
@@ -0,0 +1,31 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+/* eslint-disable camelcase */
+
+/**
+ * Check if a chart is one of the filter chart types, i.e., FilterBox
+ * and any charts with `table_filter = TRUE`.
```

Comment:
Even the chart is not a `table` viz, it can still have `table_filter = TRUE`? this sounds very problematic. Does that mean any chart can be filter?

Reply:
Yes. That was how table filter was originally implemented. I think it’s probably OK. Tableau also has this feature where you can filter/highlight by  categorical values from linked charts.

Reply:
but in the code you only have `getFilterConfigsFromTableChart` or `getFilterConfigsFromFilterBox` right? So only 2 viz types **_are eligible_** work as filter.

Reply:
For now. But the design should account for generic cases, which is why I proposed to change “table_filter” to a more generic name. Current implementation also tightly couple viz type with the filters, which kind of goes against our goal of making visualizations more pluggable. But we can deal with that one at a time.